bundles/EcommerceFrameworkBundle/CartManager/AbstractCartItem.php

```diff
@@ -94,15 +94,15 @@ public function getCount()
      */
     public function setProduct(CheckoutableInterface $product, bool $fireModified = true)
     {
-        if ((empty($product) || $this->productId != $product->getId()) && $this->getCart() && !$this->isLoading && $fireModified) {
+        if ($this->productId !== $product->getId() && !$this->isLoading && $fireModified) {
```

Comment:
why remove $this->getCart() check?

Reply:
@fashxp getChart always return CartInterface, or is this wrong?

Reply:
could be null...

Reply:
Add it again and change the getCart here: https://github.com/pimcore/pimcore/pull/9067/commits/7abdad20110baf1d3202c1d555a105a28743e408